qutebrowser/browser/webkit/downloads.py

```diff
@@ -1316,6 +1331,11 @@ def get_tmpfile(self, suggested_name):
             A tempfile.NamedTemporaryFile that should be used to save the file.
         """
         tmpdir = self._get_tmpdir()
+        encoding = sys.getfilesystemencoding()
+        suggested_name = utils.force_encoding(suggested_name, encoding)
+        # Make sure that the filename is not too long
+        if len(suggested_name) > 20:
```

Comment:
I think we can bump this up a bit - [the usual limit](http://serverfault.com/a/9548) seems to be 255 bytes, so even when assuming 5-byte UTF-8 chars for every letter, 50 should be fine.